examples/documentation_examples/examples.go

```diff
@@ -2956,3 +2957,188 @@ func StableAPIExamples() {
 	StableAPINonStrictExample()
 	StableAPIDeprecationErrorsExample()
 }
+
+func insertSnapshotQueryTestData(ctx context.Context, t *testing.T, client *mongo.Client) {
+	_, err := client.Database("pets").Collection("cats").InsertMany(ctx, []interface{}{
+		bson.D{
+			{"adoptable", false},
+			{"name", "Miyagi"},
+			{"color", "grey-white"},
+			{"age", 14},
+		},
+		bson.D{
+			{"adoptable", true},
+			{"name", "Joyce"},
+			{"color", "black"},
+			{"age", 10},
+		},
+	})
+	require.NoError(t, err)
+
+	_, err = client.Database("pets").Collection("dogs").InsertMany(ctx, []interface{}{
+		bson.D{
+			{"adoptable", true},
+			{"name", "Cormac"},
+			{"color", "rust"},
+			{"age", 7},
+		},
+		bson.D{
+			{"adoptable", true},
+			{"name", "Frank"},
+			{"color", "yellow"},
+			{"age", 2},
+		},
+	})
+	require.NoError(t, err)
+
+	_, err = client.Database("retail").Collection("sales").InsertMany(ctx, []interface{}{
+		bson.D{
+			{"shoeType", "hiking boot"},
+			{"price", 30.0},
+			{"saleDate", time.Now()},
+		},
+	})
+	require.NoError(t, err)
+}
+
+func rollbackSnapshotQueryTestData(ctx context.Context, client *mongo.Client) {
+	client.Database("pets").Drop(ctx)
+	client.Database("retail").Drop(ctx)
+}
+
+func snapshotQueryPetExample(ctx context.Context, t *testing.T, client *mongo.Client) error {
+	// Start Snapshot Query Example 1
+	// Connect to the pets database
+	db := client.Database("pets")
+
+	// Start a session with snapshotting
+	sess, err := client.StartSession(new(options.SessionOptions).SetSnapshot(true))
+	if err != nil {
+		return err
+	}
+	defer sess.EndSession(ctx)
+
+	var adoptablePetsCount int32
+	err = mongo.WithSession(ctx, sess, func(sc mongo.SessionContext) error {
+		// count the adoptable cats
+		adoptableCatsOuput := "adoptableCatsCount"
+		cursor, err := db.Collection("cats").Aggregate(sc, mongo.Pipeline{
+			bson.D{{"$match", bson.D{{"adoptable", true}}}},
+			bson.D{{"$count", adoptableCatsOuput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(sc)
+		resp, err := cursor.Current.LookupErr(adoptableCatsOuput)
+		if err != nil {
+			return err
+		}
+		adoptableCatsCount, _ := resp.Int32OK()
```

Comment:
I don't think we want to give example code that throws away the `ok` value here. Could you add something like
```
if !ok {
    return fmt.Errorf("expected %q field to be of type Int32, got %T", adoptableCatsOutput, resp)
}
```

Reply:
It's actually safe to call `Lookup(adoptableCatsOutput).Int32OK()` because `Lookup` returns an empty `RawValue`, so `Int32OK` will return `false` if either the target field is missing _or_ if it's not an int32. Also, `resp` is always a `RawValue`, so the "%T" formatter will print `bson.RawValue`. What you want is `resp.Type`. If we use `Lookup` instead of `LookupErr`, we won't have that intermediary value so we can just print the ExtJSON representation of the whole doc in the error.

E.g.
```go
adoptableCatsCount, ok := cursor.Current.Lookup(adoptableCatsOutput).Int32OK()
if !ok {
	return fmt.Errorf("failed to find int32 field %q in document %v", adoptableCatsOutput, cursor.Current)
}
```